cms/djangoapps/contentstore/management/commands/swap_from_auto_track_cohort_pilot.py

```diff
@@ -0,0 +1,192 @@
+            items_to_update = []
+            if not items:
+                raise CommandError("Items for Course with key '%s' not found." % rerun_course_key)
+
+            for item in items:
+                # Verify that there exists group access for this xblock, otherwise skip these checks
+                if item.group_access:
+                    set_audit_enrollment_track = False
+                    set_verified_enrollment_track = False
+
+                    # Check the partition and group IDs for the audit course groups, if they exist in
+                    # the xblock's access settings then set the audit track flag to true
+                    for audit_course_user_group_partition_group in random_audit_course_user_group_partition_groups:
+                        audit_partition_group_access = item.group_access.get(
+                            audit_course_user_group_partition_group.partition_id,
+                            None
+                        )
+                        if (audit_partition_group_access
+                                and audit_course_user_group_partition_group.group_id in audit_partition_group_access):
+                            set_audit_enrollment_track = True
+
+                    # Check the partition and group IDs for the verified course group, if it exists in
+                    # the xblock's access settings then set the verified track flag to true
+                    verified_partition_group_access = item.group_access.get(
+                        verified_course_user_group_partition_group.partition_id,
+                        None
+                    )
+                    if (verified_partition_group_access
+                            and verified_course_user_group_partition_group.group_id in verified_partition_group_access):
+                        set_verified_enrollment_track = True
+
+                    # Add the enrollment track ids to a group access array
+                    enrollment_track_group_access = []
+                    if set_audit_enrollment_track:
+                        enrollment_track_group_access.append(settings.COURSE_ENROLLMENT_MODES['audit'])
+                    if set_verified_enrollment_track:
+                        enrollment_track_group_access.append(settings.COURSE_ENROLLMENT_MODES['verified'])
+
+                    # If either the audit track, or verified track needed an update, set the access, update and publish
+                    if set_verified_enrollment_track or set_audit_enrollment_track:
+                        # Sets whether or not an xblock has changes
+                        has_changes = module_store.has_changes(item)
+
+                        # Check that the xblock does not have changes and add it to be updated, otherwise add an error
+                        if not has_changes:
+                            item.group_access = {ENROLLMENT_TRACK_PARTITION_ID: enrollment_track_group_access}
```

Comment:
We could potentially be losing some group_access settings here. Imagine that an item has group_access set to a "audit" and/or "verified" content group, but also has group_access set to ANOTHER content group (not linked to cohorts at all). Since we are enforcing (via UI only) that an item has differentiate by either Enrollment Tracks or Content Groups, I think we should append an error in that case instead of just losing the additional content groups. Will probably need to do this check up in the region of 106-118.

Let me know if this isn't clear and I can talk through an example.